### ISSUE TITLE
remove() instead of element() in ServiceWatcherUtil

### ISSUE THREAD

**Issue by EmmanuelMess** [TeamAmaze/AmazeFileManager#1372]
Shouldn't the call be remove() instead of element() to actually remove the head of the queue?:
https://github.com/TeamAmaze/AmazeFileManager/blob/b775042ca900102918d725efcba54c0a864feddd/app/src/main/java/com/amaze/filemanager/utils/ServiceWatcherUtil.java#L207

Related #1306

**Comment by EmmanuelMess** [comment]
Ok, this won't work because when the process finishes it is removed from the queue (again), causing a crash (with the last process). Ideas to make the process start only once?

**Comment by VishalNehra** [comment]
Processes do start only once. Any proof that processes start multiple times?

**Comment by EmmanuelMess** [comment]
1. Put a breakpoint before the service starts the background thread (B1) (task) and another one after it starts (B2)
2. Copy something
3. The exectution will stop in B1, continue, the execution will stop in B2, don't continue.
4. The execution will stop in B1 again (B2 is in another thread), indicating that the service started twice.

**Comment by VishalNehra** [comment]
BTW about this discussion, service entry is removed after it is finished, in the ````ServiceWatcherUtil#watch````. You shouldn't call remove in place of element.

**Comment by EmmanuelMess** [comment]
> Just put a simple Log.d in anywhere in onStartCommand  of service, it is printed only once in logs. If it were started  multiple times it would've been printed multiple times too, because that  is where we create the notification.

* Testing on my Moto E Gen 2 4G with master and see 3 lines (copying a single 81MB file in my sd):
```
07-29 15:41:05.303 com.amaze.filemanager D/Copy service: Started copy service with intent: Intent { cmp=com.amaze.filemanager/.asynchronous.services.CopyService (has extras) }
07-29 15:41:05.378 com.amaze.filemanager D/Copy service: Started copy service with intent: Intent { cmp=com.amaze.filemanager/.asynchronous.services.CopyService (has extras) }
07-29 15:41:05.469 com.amaze.filemanager D/Copy service: Started copy service with intent: Intent { cmp=com.amaze.filemanager/.asynchronous.services.CopyService (has extras) }
```

* Copying 1001 files 0B files in sd I didn't see more than one line, although the notification was missing (also, files seem to be copied at 1 every ~4s, wut?)... and it crashed (after 179 files):
```
07-29 15:50:24.365 com.amaze.filemanager E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.amaze.filemanager, PID: 31173
    java.lang.RuntimeException: Unable to start service com.amaze.filemanager.asynchronous.services.CopyService@33ad6bc with null: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.content.Intent.getBooleanExtra(java.lang.String, boolean)' on a null object reference
        at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3045)
        at android.app.ActivityThread.-wrap17(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1452)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:148)
        at android.app.ActivityThread.main(ActivityThread.java:5443)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.content.Intent.getBooleanExtra(java.lang.String, boolean)' on a null object reference
        at com.amaze.filemanager.asynchronous.services.CopyService.onStartCommand(CopyService.java:107)
        at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3028)
        at android.app.ActivityThread.-wrap17(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1452) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:148) 
        at android.app.ActivityThread.main(ActivityThread.java:5443) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618) 
```
* Retying the 1001 file copy, no notifications is shown and it is really slow, (around 1hr to copy 860, 0B, files), when I opened the cronometer app by accident Amaze crashed:
```
07-29 16:58:38.495 com.amaze.filemanager E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.amaze.filemanager, PID: 6041
    java.lang.RuntimeException: Unable to start service com.amaze.filemanager.asynchronous.services.CopyService@33ad6bc with null: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.content.Intent.getBooleanExtra(java.lang.String, boolean)' on a null object reference
        at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3045)
        at android.app.ActivityThread.-wrap17(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1452)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:148)
        at android.app.ActivityThread.main(ActivityThread.java:5443)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.content.Intent.getBooleanExtra(java.lang.String, boolean)' on a null object reference
        at com.amaze.filemanager.asynchronous.services.CopyService.onStartCommand(CopyService.java:107)
        at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3028)
        at android.app.ActivityThread.-wrap17(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1452) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:148) 
        at android.app.ActivityThread.main(ActivityThread.java:5443) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618) 
```


<br/>
<br/>

> BTW about this discussion, service entry is removed after it is finished, in the ````ServiceWatcherUtil#watch````. You shouldn't call remove in place of element. 

[Yep](https://github.com/TeamAmaze/AmazeFileManager/issues/1372#issuecomment-406407903), didn't notice when creating the issue.

**Comment by VishalNehra** [comment]
1001 files with 0B size? Am I missing something?
BTW if that's the case I'll look into the code since I wrote this. You can
create a new issue for this and assign to me.

**Comment by VishalNehra** [comment]
> BTW about this discussion, service entry is removed after it is finished, in the ServiceWatcherUtil#watch. You shouldn't call remove in place of element.

Closing
